battle-service/src/main/java/by/ibondarau/tictactoe/battleservice/service/impl/BattleServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+package by.ibondarau.tictactoe.battleservice.service.impl;
+
+import by.ibondarau.tictactoe.battleservice.dao.BattleDao;
+import by.ibondarau.tictactoe.battleservice.exception.BusinessException;
+import by.ibondarau.tictactoe.battleservice.exception.NotFoundException;
+import by.ibondarau.tictactoe.battleservice.model.Battle;
+import by.ibondarau.tictactoe.battleservice.model.BattleResult;
+import by.ibondarau.tictactoe.battleservice.model.BattleStatus;
+import by.ibondarau.tictactoe.battleservice.model.FirstMoveRule;
+import by.ibondarau.tictactoe.battleservice.model.Move;
+import by.ibondarau.tictactoe.battleservice.service.BattleService;
+import by.ibondarau.tictactoe.battleservice.util.BattleUtils;
+import by.ibondarau.tictactoe.battleservice.util.RandomUtils;
+import com.sun.istack.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Service
+public class BattleServiceImpl implements BattleService {
+
+    private final BattleDao battleDao;
+    private final RandomUtils randomUtils;
+
+    private final BattleUtils battleUtils;
+
+    @Autowired
+    public BattleServiceImpl(BattleDao battleDao, RandomUtils randomUtils, BattleUtils battleUtils) {
```

Comment:
not critical . but i prefer use lombok with `@RequiredArgsConstructor `to inject beans.